src/main/java/com/karankumar/bookproject/backend/service/BookService.java

```diff
@@ -57,13 +66,24 @@ public void save(Book book) {
     private boolean bookHasAuthorAndPredefinedShelf(Book book) {
         return book != null && book.getAuthor() != null && book.getPredefinedShelf() != null;
     }
+    
+    private boolean bookHasPublisher(Book book) {
+    	return book.getPublisher() != null;
+    }
```

Comment:
Don't think we need a separate method for one condition

Reply:
@amit1307 I think this is more readable